python/jupytergis_lab/jupytergis_lab/notebook/gis_document.py

```diff
@@ -138,6 +143,59 @@ def _export_to_qgis(self, path: str | Path) -> bool:
 
         return export_project_to_qgis(path, virtual_file)
 
+    def add_layer(
```

Comment:
This function is confusing to me. It seems to only allow 2 kinds of layers, but has a generic name.
I wonder if we should add this function in a follow up PR, with support to others kinds of layer, to make it really generic.

Reply:
I had the same gut feeling 👍🏽

Reply:
I understand your concern about this! I waffled about this myself, but figured if I'm going to write this function I might as well let users benefit from it if they like, and also enable the community to help increase the function's scope over time. I personally thought it's not a big deal since the docs make it clear that this helper method currently only works for a subset of layer types, and the `NotImplementedError` makes the limitations clear at runtime.

I think you both know better than me, however! I'm going to move this to a "private" function in `explore.py` for now, and we can come back to it later. If my thoughts above have swayed you,  we can revert the change before merge. Up to y'all! :)

Reply:
119c281